var/spack/repos/builtin/packages/py-sonata-network-reduction/package.py

```diff
@@ -24,21 +25,28 @@ class PySonataNetworkReduction(PythonPackage):
     depends_on('py-setuptools', type=('build', 'run'))
 
     depends_on('py-numpy@1.14:', type='run')
-    depends_on('py-h5py@2.10:', type='run')
-    depends_on('py-pandas@0.25:', type='run')
-    depends_on('py-lxml@4.3.4:', type='run')
-    depends_on('py-pyyaml@5.3:', type='run')
-    depends_on('py-tqdm@4.34:', type='run')
-    depends_on('py-click@6.7:', type='run')
-    depends_on('py-joblib@0.14:', type='run')
+    depends_on('py-pandas@0.25:1.99', type='run')
+    depends_on('py-lxml@4.3.4:4.99', type='run', when='@:0.1.2')
+    depends_on('py-pyyaml@5.3:5.99', type='run')
+    depends_on('py-tqdm@4.34:4.99', type='run')
+    depends_on('py-click@6.7:7.99', type='run')
+    depends_on('py-joblib@0.14:0.99', type='run', when='@:0.1.1')
     depends_on('py-bluepyopt@1.8.68:', type='run')
-    depends_on('py-bglibpy@4.3:', type='run')
-    depends_on('py-morphio@2.3.9:', type='run')
+    depends_on('py-bglibpy@4.3:4.999', type='run')
+    depends_on('py-morphio@2.3.9:2.999', type='run')
```

Comment:
Is this already a known constraint?

Reply:
Now I restrict all dependencies by next major version for safety. Here is the corresponding [setup.py](https://bbpcode.epfl.ch/browse/code/nse/sonata-network-reduction/tree/setup.py)